File: TrainWise.Web/Models/DatasetSummary.cs
namespace TrainWise.Web.Models;

public sealed class DatasetSummary
{
    public string DatasetId { get; set; } = string.Empty;
    public int RowCount { get; set; }
    public int ColumnCount { get; set; }
    public Dictionary<string, string> ColumnTypes { get; set; } = new();
    public Dictionary<string, MissingValueInfo> MissingValues { get; set; } = new();
    public Dictionary<string, SummaryStats> SummaryStats { get; set; } = new();
    public Dictionary<string, int>? ClassDistribution { get; set; }
    public List<List<double>> CorrelationMatrix { get; set; } = new();
    public double? ImbalanceRatio { get; set; }

    // Computed properties for UI convenience
    public List<string> Columns => ColumnTypes.Keys.ToList();
    public List<List<string>> Preview { get; set; } = new();
}

public sealed class MissingValueInfo
{
    public int Count { get; set; }
    public double Percentage { get; set; }
}

public sealed class SummaryStats
{
    public double Mean { get; set; }
    public double Median { get; set; }
    public double Std { get; set; }
    public double Min { get; set; }
    public double Max { get; set; }
}

File: TrainWise.MLService/app/models/dataset_summary.py
from typing import Dict, List, Optional
from pydantic import BaseModel

class MissingValueInfo(BaseModel):
    count: int
    percentage: float

class SummaryStats(BaseModel):
    mean: float
    median: float
    std: float
    min: float
    max: float

class DatasetSummary(BaseModel):
    datasetId: str
    rowCount: int
    columnCount: int
    columnTypes: Dict[str, str]
    missingValues: Dict[str, MissingValueInfo]
    summaryStats: Dict[str, SummaryStats]
    classDistribution: Optional[Dict[str, int]] = None
    correlationMatrix: List[List[float]]
    imbalanceRatio: Optional[float] = None
